doc/source/overview/faq.md

```diff
@@ -0,0 +1,56 @@
+ - The predicted class of the data point does not change regardless of the perturbations applied to it.
+ - The predicted class of the data point always changes regardless of the perturbations applied to it.
+
+**Q. Why is my anchor explanation so long (tabular or text data) or covers much of the image (image data)?**
+
+**A.** This is expected behaviour and can happen in two ways:
+ - The data point to be explained lies near the decision boundary of the classifier. Thus, many more predicates are needed to ensure that a data point keeps the predicted class as small changes to the feature values may push the prediction to another class.
+ - For tabular data, sampling perturbations is done using a training set. If the training set is imbalanced, explaining a minority class data point will result in oversampling perturbed features typical of majority classes so the algorithm would struggle finding a short anchor exceeding the specified precision level. For a concrete example, see [Anchor explanations for income prediction
+](../examples/anchor_tabular_adult.ipynb).
+
+## Counterfactual explanations
+
+**Q. I'm trying to use the methods [Counterfactual](../methods/CF.ipynb), [CounterfactualProto](../methods/CFProto.ipynb) or [CEM](../methods/CEM.ipynb) on a tree-based model such as decision trees, random forests,  or gradient boosted models (e.g. `xgboost`) but not finding any counterfactual examples, what's wrong?**
+
+**A.** These methods only work on a subset of black-box models, namely ones whose decision function is differentiable with respect to the input data and hence amenable to gradient-based counterfactual search. Since for tree-based models the decision function is piece-wise constant these methods won't work. It is recommended to use [CFRL](../methods/CFRL.ipynb) instead.
+
+**Q. I'm getting an error trying to use the methods [Counterfactual](../methods/CF.ipynb), [CounterfactualProto](../methods/CFProto.ipynb) or [CEM](../methods/CEM.ipynb), especially if trying to use one of these methods together with [IntegratedGradients](../methods/IntegratedGradients.ipynb) or [CFRL](../methods/CFRL.ipynb), what's happening?**
+
+**A.** At the moment the 3 counterfactual methods are implemented using TensorFlow 1.x constructs. This means that when running these methods, first we need to disable behaviour specific to TensorFlow 2.x as follows:
+
+```ipython3
+import tensorflow as tf
+tf.compat.v1.disable_v2_behavior()
+```
+Unfortunately, running this line means it's impossible to run explainers based on TensorFlow 2.x such as [IntegratedGradients](../methods/IntegratedGradients.ipynb) or [CFRL](../methods/CFRL.ipynb). Thus at the moment it is impossible to run these explainers together in the same Python interpreter session. Ultimately the fix is to rewrite the TensorFlow 1.x implementations in idiomatic TensorFlow 2.x and [some work has been done](https://github.com/SeldonIO/alibi/pull/403) but is currently not prioritised.
```

Comment:
Thus at the moment it is impossible -> Thus at the moment, it is impossible

Correction suggested by Grammarly